br/cmd/tidb-lightning/main.go

```diff
@@ -89,12 +90,21 @@ func main() {
 		return app.RunOnce(context.Background(), cfg, nil)
 	}()
 
+	finished := true
+	if common.IsContextCanceledError(err) {
+		err = nil
+		finished = false
+	}
 	if err != nil {
 		logger.Error("tidb lightning encountered error stack info", zap.Error(err))
 		fmt.Fprintln(os.Stderr, "tidb lightning encountered error: ", err)
 	} else {
-		logger.Info("tidb lightning exit")
-		fmt.Fprintln(os.Stdout, "tidb lightning exit")
+		logger.Info("tidb lightning exit", zap.Bool("finished", finished))
+		exitMsg := "tidb lightning exit successfully"
+		if finished {
```

Comment:
a bit weird😂
https://github.com/pingcap/tidb/blob/8da8863459e753f8ccc93694160672c05ec9b959/br/cmd/tidb-lightning/main.go#L105-L110

Reply:
it's fixed in https://github.com/pingcap/tidb/pull/30908/files#diff-f7e9e372818f419eeab8024ccb8a670a07c9f07768d492ed95c9f0ee14ab0d5e, but that pr didn't pick to 5.3 too, we fix it here